src/mixins/java/org/spongepowered/common/mixin/inventory/event/world/inventory/AbstractContainerMenuMixin_Inventory.java

```diff
@@ -162,6 +164,18 @@ public abstract class AbstractContainerMenuMixin_Inventory implements TrackedCon
 
     }
 
+    @Inject(method = "broadcastFullState", at = @At("HEAD"))
+    private void impl$broadcastFullStateWithTransactions(final CallbackInfo ci) {
+        if (!PhaseTracker.SERVER.onSidedThread()) {
+            return;
+        }
+        this.bridge$detectAndSendChanges(false, false);
+        this.impl$broadcastDataSlots();
```

Comment:
You forgot to add the flag for `shadow$synchronizeDataSlotToRemote` that we discussed.